src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,533 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+        """Label rows as headers, notes, or totals."""
+        # Define header qualifications
+        header_strings = [
+            "hydro",
+            "hyrdo",
+            "internal",
+            "wind",
+            "solar",
+            "gas",
+            "diesel",
+            "diesal",
+            "steam",
+            "other",
+            "combustion",
+            "combustine",
+            "fuel cell",
+            "hydraulic",
+            "waste",
+            "landfill",
+            "photovoltaic",
+            "nuclear",
+            "oil",
+            "renewable",
+            "facilities",
+            "combined cycle",
+        ]
+        nonheader_strings = [
+            "#",
+            r"\*",
+            "pg",
+            "solargenix",
+            "solargennix",
+            r"\@",
+            "rockton",
+            "albany steam",
+        ]
+        exceptions = [
+            "hydro plants: licensed proj. no.",
+            "hydro license no.",
+            "hydro: license no.",
+            "hydro plants: licensed proj no.",
+        ]
+        not_header_if_cols_na = [
+            "construction_year",
+            "net_generation_mwh",
+            "total_cost_of_plant",
+            "capex_per_mw",
+            "opex_total",
+            "opex_fuel",
+            "opex_maintenance",
+            "fuel_cost_per_mmbtu",
+        ]
+
+        def label_header_rows(df: pd.DataFrame) -> pd.DataFrame:
+            """Label header rows.
+
+            This function labels rows it believes are possible headers based on whether
+            they contain information in certain key columns. Of those possible headers,
+            ones that contain a specific key word or phrase are dubbed headers. Leftover
+            possible header rows are evaluated as possible note rows in the
+            _label_notes_rows() function.
+            """
+            logger.info(f"{self.table_id.value}: Labeling header rows")
+
+            # Label possible header rows (based on the nan cols specified above)
+            df.loc[
+                df.filter(not_header_if_cols_na).isna().all(1), "possible_header"
+            ] = True
+
+            # Label good header rows (based on whether they contain key strings)
+            possible_header = df["possible_header"]
+            good_header = df["plant_name_ferc1"].str.contains("|".join(header_strings))
+            not_bad = ~df["plant_name_ferc1"].str.contains("|".join(nonheader_strings))
+
+            df.loc[possible_header & good_header & not_bad, "row_type"] = "header"
+            df.loc[df["plant_name_ferc1"].isin(exceptions), "row_type"] = "header"
+
+            return df
+
+        def label_total_rows(df: pd.DataFrame) -> pd.DataFrame:
+            """Label total rows."""
+            logger.info(f"{self.table_id.value}: Labeling total rows")
+
+            df.loc[df["plant_name_ferc1"].str.contains("total"), "row_type"] = "total"
```

Comment:
Yes. There's only one instance (right now) where a header gets overridden as a total, and it's correct.

Reply:
Here's what it looks like in the data. The "total" rows are at the bottom. 

|       | plant_name_ferc1                  |
|------:|:----------------------------------|
| 17879 | solar plants                      |
| 17880 | la senita                         |
| 17881 | rio rico                          |
| 17882 | jacobson                          |
| 17883 | total                             |
| 17884 | total solar operation/maintenance |

Reply:
maybe this is neurotic but it seems confusing that  `total solar operation/maintenance` could be labeled as a header in one step and then overwritten in another. could we add a `~df["plant_name_ferc1"].str.contains("total")` into the labeling of the headers?

This may not work in all in cases in which your re-labeling totals. So it might be better to re-brand the labeling of headers as labeling possible headers... which your also doing but idk maybe its an aggressive header label-er?

Reply:
Yeah I see what you mean. It was very much designed to start off vague and then specify what type of row it actually is. Maybe the right thing to do here is to have the function that labels possible headers be called something like find extraneous rows or something. 

Then, maybe it makes sense to run the total row call first.

Reply:
Ok, I started by separating out the `possible_headers` label and changing it to `possible_header_or_note`. I'll take a closer look at the nature of overriding headers etc. once I've looked through some of the other comments.